src/main/java/org/spongepowered/mod/mixin/core/world/MixinWorld.java

```diff
@@ -308,6 +315,39 @@ public boolean spawnEntity(Entity entity) {
     }
 
     @Override
+    public boolean spawnEntity(Entity entity) {
+        checkNotNull(entity, "Entity cannot be null!");
+        return spawnEntityInWorld(((net.minecraft.entity.Entity) entity));
+    }
+
+    // Copy of original except doesn't force loading if entity is a HumanEntity
+    @Overwrite
+    public boolean spawnEntityInWorld(net.minecraft.entity.Entity entityIn) {
+        if (!this.isRemote && (entityIn == null || (entityIn instanceof net.minecraft.entity.item.EntityItem && this.restoringBlockSnapshots))) {
+            return false;
+        }
+        int cx = MathHelper.floor_double(entityIn.posX / 16.0D);
```

Comment:
Bit-shifting doubles is usually a bad idea. You'd usually shift if it were a long or integer coord.